.github/workflows/main.yml

```diff
@@ -170,6 +170,12 @@ jobs:
       run: bundle exec rake lex:topgems
     - name: Parse Top 100 Gems
       run: bundle exec rake parse:topgems
+    - name: Serialized size stats with all fields
```

Comment:
This is an interesting metric, but I don't think we want it to run on every CI run. The `topgems` check is already one of the longest-running. I'm fine keeping the task around though.

Reply:
This only takes like 2 seconds to execute, so I think it has no observable impact on CI time.

Reply:
On this CI run https://github.com/ruby/yarp/actions/runs/6207594674/job/16852941189 it's 3+19+3 seconds.

Reply:
I wonder if `rake lex:topgems` could be optimized further, that takes 4m22s in that run.

Reply:
A quick profile with stackprof/autorun gives:
```
stackprof /tmp/stackprof20230919-104519-jjlzxp.dump
==================================
  Mode: wall(1000)
  Samples: 21520 (90.63% miss rate)
  GC: 13603 (63.21%)
==================================
     TOTAL    (pct)     SAMPLES    (pct)     FRAME
      8441  (39.2%)        8441  (39.2%)     (marking)
      7520  (34.9%)        7520  (34.9%)     (sweeping)
      5530  (25.7%)        5530  (25.7%)     Thread#join
         3   (0.0%)           3   (0.0%)     Kernel#require
     15963  (74.2%)           2   (0.0%)     (garbage collection)
      5530  (25.7%)           0   (0.0%)     Array#map
      7890  (36.7%)           0   (0.0%)     YARP.parallelize
...
```
Which looks strange, like non-main threads are not included or so.
That does seem like a lot of GC though.
Also `ENV.fetch("WORKERS") { 16 }` is probably way too high for GitHub Actions, maybe it could use `Etc.nprocessors` or so, or it might be more efficient to not parallelize that at all on CRuby (might just increase GVL contention).

Reply:
@kddnewton I can move this to a separate CI job, then it won't make this one any longer. Would that address your concern?